lib/ad_service/query/time_management.ex

```diff
@@ -33,6 +54,22 @@ defmodule AdService.Query.TimeManagement do
     end
   end
 
+  defp build_allowed_on_weekends_query(query, day_of_week_int) when day_of_week_int < 6, do: query
+
+  defp build_allowed_on_weekends_query(query, _) do
+    query
+    |> where([_creative, campaign, ...], campaign.weekdays_only == false)
+  end
+
+  defp local_time_zone_check(function, query, ip_address) do
+    with {:ok, %{time_zone: time_zone}} when time_zone != "" <-
+           ip_address |> Framework.Geolocation.find_by_ip(:city) do
```

Comment:
I want pipes 😭

Reply:
@hopsoft yeah maybe, if we're doing any campaign analytics where we query that, i'd say probably. Alternatively, we could pull it into some of the materialized views we have that are built for speeding up campaign analytics queries (tldr minimizing joins where otherwise they'd be there to slow down queries).

Reply:
Is this similar @mbeacom? https://toolz.readthedocs.io/en/latest/api.html#toolz.functoolz.pipe

Reply:
I'm not too familiar with the `toolz` module.  I'm interested to profile that and see how the performance is, though!